pretext/project/__init__.py

```diff
@@ -341,25 +338,60 @@ def build(
                 )
 
             log.info(f"Preparing to build into {self.output_dir_abspath()}.")
-
-            build.build(
-                self.format,
-                self.source_abspath(),
-                self.publication_abspath(),
-                self.output_dir_abspath(),
-                self.output_filename,
-                self.stringparams,
-                custom_xsl=custom_xsl,
-                xmlid=xmlid,
-                zipped=self.compression is not None,
-                project_path=self._project.abspath(),
-                latex_engine=self.latex_engine,
-                executables=self._project._executables.dict(),
-                # TODO: what if this isn't defined? Should we have a default in the project file instead?
-                braille_mode=self.braille_mode,
-            )
+            if self.format == "html":
+                core.html(
+                    xml=self.source_abspath(),
+                    pub_file=self.publication_abspath().as_posix(),
+                    stringparams=self.stringparams,
+                    xmlid_root=xmlid,
+                    file_format=self.compression or "html",
+                    extra_xsl=custom_xsl,
+                    out_file=None,
```

Comment:
If we're making a compressed HTML file, shouldn't we allow the `output_file` to determine the name of the resulting .zip archive?

Reply:
I'm not sure what happens in core if both a `dest_dir` and `out_file` are specified.  I suppose that isn't our problem though.

Reply:
Also, the `@validator` for `output_filename` would need to change in the case that compression is set.

Reply:
This should be fixed now.  Please take a look, @bjones1